test/tests/desktop/credentials/DesktopCredentialsFacadeTest.ts

```diff
@@ -0,0 +1,153 @@
+import o from "@tutao/otest"
+
+import path from "node:path"
+import { DesktopNativeCredentialsFacade, resolveChecked } from "../../../../src/desktop/credentials/DesktopNativeCredentialsFacade.js"
+import { DesktopNativeCryptoFacade } from "../../../../src/desktop/DesktopNativeCryptoFacade.js"
+import { CredentialEncryptionMode } from "../../../../src/misc/credentials/CredentialEncryptionMode.js"
+import { makeKeyStoreFacade } from "../../TestUtils.js"
+import { assertThrows } from "@tutao/tutanota-test-utils"
+import { function as fn, matchers, object, verify, when } from "testdouble"
+import { LanguageViewModel } from "../../../../src/misc/LanguageViewModel.js"
+import { DesktopConfig } from "../../../../src/desktop/config/DesktopConfig.js"
+import { defer, delay, stringToBase64 } from "@tutao/tutanota-utils"
+import { DesktopConfigKey } from "../../../../src/desktop/config/ConfigKeys.js"
+import { CommonNativeFacade } from "../../../../src/native/common/generatedipc/CommonNativeFacade.js"
+import { CancelledError } from "../../../../src/api/common/error/CancelledError.js"
+import { KeyPermanentlyInvalidatedError } from "../../../../src/api/common/error/KeyPermanentlyInvalidatedError.js"
+
+async function loadArgon2ModuleFromFile(path: string): Promise<WebAssembly.Exports> {
+	if (typeof process !== "undefined") {
+		try {
+			const { readFile } = await import("node:fs/promises")
+			const wasmBuffer = await readFile(path)
+			return (await WebAssembly.instantiate(wasmBuffer)).instance.exports
+		} catch (e) {
+			throw new Error(`Can't load argon2 module: ${e}`)
+		}
+	} else {
+		return (await WebAssembly.instantiateStreaming(await fetch(path))).instance.exports
+	}
+}
+
+o.spec("DesktopNativeCredentialsFacade", () => {
+	const key = new Uint8Array([1, 2, 3])
+	const keyStoreFacade = makeKeyStoreFacade(key)
+
+	const getSubject = async () => {
```

Comment:
why do we do this instead of usual `o.beforeEach()`?

Reply:
I guess because it's old magic code

Reply:
fair